pkg/cloudprovider/aws/instancetypes_test.go

```diff
@@ -220,27 +220,55 @@ var _ = Describe("Instance Types", func() {
 		Expect(nodeNames.Len()).To(Equal(2))
 	})
 	It("should set pods to 110 if not using ENI-based pod density", func() {
-		opts.AWSENILimitedPodDensity = false
+		optsCopy := opts
+		optsCopy.AWSENILimitedPodDensity = false
 		instanceInfo, err := instanceTypeProvider.getInstanceTypes(ctx)
 		Expect(err).To(BeNil())
 		provisioner = test.Provisioner()
 		for _, info := range instanceInfo {
-			it := NewInstanceType(injection.WithOptions(ctx, opts), info, provisioner.Spec.KubeletConfiguration, "", provider, nil)
+			it := NewInstanceType(injection.WithOptions(ctx, optsCopy), info, provisioner.Spec.KubeletConfiguration, "", provider, nil)
 			resources := it.Resources()
 			Expect(resources.Pods().Value()).To(BeNumerically("==", 110))
 		}
 	})
 	It("should not set pods to 110 if using ENI-based pod density", func() {
-		opts.AWSENILimitedPodDensity = true
+		optsCopy := opts
```

Comment:
Setting up these copies was attempt to work around some apparent fragility.  I think another test in the suite is directly modify the opts fixture here.